components/generic_modal.tsx

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React from 'react';
+import {Modal} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+
+import './generic_modal.scss';
+
+type Props = {
+    onHide: () => void;
+    modalHeaderText: string | JSX.Element;
+    show?: boolean;
+    handleCancel?: () => void;
+    handleConfirm?: () => void;
+    confirmButtonText?: string | JSX.Element;
+    confirmButtonClassName?: string;
+    cancelButtonText?: string | JSX.Element;
+    isConfirmDisabled?: boolean;
+};
+
+type State = {
+    show: boolean;
+}
+
+export default class GenericModal extends React.PureComponent<Props, State> {
+    static defaultProps: Partial<Props> = {
+        show: true,
+    };
+
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            show: props.show!,
+        };
+    }
+
+    onHide = () => {
+        this.setState({show: false}, this.props.onHide);
+    }
+
+    handleCancel = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+        event.preventDefault();
+        this.onHide();
+        if (this.props.handleCancel) {
+            this.props.handleCancel();
+        }
+    }
+
+    handleConfirm = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+        event.preventDefault();
+        this.onHide();
+        if (this.props.handleConfirm) {
+            this.props.handleConfirm();
+        }
+    }
+
+    render() {
+        let confirmButton;
+        if (this.props.handleConfirm) {
+            let confirmButtonText = (
+                <FormattedMessage
+                    id='generic_modal.confirm'
+                    defaultMessage='Confirm'
+                />
+            );
+            if (this.props.confirmButtonText) {
+                confirmButtonText = (
+                    <React.Fragment>
```

Comment:
Are these fragments necessary or can we just do `confirmButtonText = this.props.confirmButtonText`?

Reply:
I also wanted to ask about the reasoning behind using the React.Fragment, the common use case is for many children rendering, but seems like this is only one button element?

Reply:
Apparently not, fixed.

Reply:
Since I allow `confirmButtonText` to be a `JSX.Element` or a `string`, it needs to be wrapped this way or TS will complain.